craft_application/application.py

```diff
@@ -239,40 +239,11 @@ def get_project(self, project_dir: pathlib.Path | None = None) -> models.Project
         self.__project = self.app.ProjectClass.from_yaml_data(yaml_data, project_path)
         return self.__project
 
-    @property
+    @cached_property
     def project(self) -> models.Project:
-        """Get this application's Project metadata.
-
-        DEPRECATED: use ``get_project`` instead.
-        """
-        warnings.warn(
-            "Direct access to an application's project property is deprecated "
-            "and will be removed in craft-application 2.0. Use get_project instead.",
-            DeprecationWarning,
-            stacklevel=1,
-        )
+        """Get this application's Project metadata."""
         return self.get_project()
```

Comment:
this is fine for now but heads-up that we'll have to review this [once get_project() gets parameters](https://github.com/canonical/craft-application/pull/150/files#diff-c395b65ed7321f1bbf3c026ade9775f6a47275ae4051f4981b0376c1730c25d4R184)